pydatastructs/miscellaneous_data_structures/queue.py

```diff
@@ -122,3 +127,50 @@ def __str__(self):
         for i in range(self.front, self.rear + 1):
             _data.append(self.items._data[i])
         return str(_data)
+
+class PriorityQueue(Queue):
+
+    '''
+    PARAMETERS
+    ==========
+    insert : method()
+    Used to insert data into the queue
+    '''
+    def __new__(cls):
+        obj = object.__new__(cls)
+        obj.queue =[]
+        return obj
+
+    def __str__(self):
+        return ''.join([str(i) for i in self.queue])
+
+    def insert(self,data):
+        self.queue.append(data)
+
+    def deleteHighestPriority(self,priority="max"):
+        if self.is_empty:
+            raise ValueError("Queue is Empty")
+        if priority=="max":
+            max_ = max(self.queue)
```

Comment:
cant we use heap to do this? checking the max every time deleteHighestPriority is called seems to be an O(n) job using a heap can reduce it to O(log(n))

Reply:
Yes as I have mentioned in my PR i am working on Binary Heaps for Priority Queue . This is one of the naive approaches for Priority Queue .

Reply:
Yes as I have mentioned in my PR i am working on Binary Heaps for Priority Queue . This is one of the naive approaches for Priority Queue .